nestkernel/simulation_manager.cpp

```diff
@@ -551,6 +551,15 @@ nest::SimulationManager::prepare_simulation_()
 
   kernel().model_manager.create_secondary_events_prototypes();
 
+  if ( kernel().mpi_manager.get_num_processes() > 1 )
+  {
+    // Check if any MPI process uses waveform relaxation and set
+    // kernel().node_manager.any_node_uses_wfr() correspondingly
+    // across all MPI processes
+    kernel().node_manager.set_any_node_uses_wfr( kernel().mpi_manager.any_true(
+      kernel().node_manager.any_node_uses_wfr() ) );
+  }
```

Comment:
This code is less legible than I had hoped. Especially, it is not really clear that we exchange local information to get a global picture. I wonder if it would not be better to replace the entire `if` block with a single call
```c++
kernel().node_manager.check_wfr_use();
```
and move the actual check to the `NodeManager` class, replacing `set_any_node_uses_wfr()` with `check_wfr_use()`, which would just be
```c++
{
  wfr_is_used_ = kernel().mpi_manager.any_true( wfr_is_used_ );
}
```
I would also change the `any_node_uses_wfr` variable and method names to `wfr_is_used`, which I think is a little clearer, and would move the check for multiple processes into the `any_true()` method.